shared/views/RoomDirectoryView.js

```diff
@@ -17,13 +20,39 @@ class RoomDirectoryView extends TemplateView {
       vm.setSearchTerm(searchInputValueBeforeRendering);
     }
 
+    function toggleSafeSearchLabel() {
+      var text = document.getElementById('checkBox_label');
+      if (text.innerHTML === 'Safe Search is on') {
+        text.innerHTML = 'Safe Search is off';
+      } else {
+        text.innerHTML = 'Safe Search is on';
+      }
+    }
+
     const roomList = new ListView(
       {
         className: 'RoomDirectoryView_roomList',
         list: vm.rooms,
         parentProvidesUpdates: false,
       },
       (room) => {
+        let name = room.name;
+        let topic = room.topic;
+        const contains = NSFW_KEYWORDS.some((element) => {
+          if (
+            name
+              .toLowerCase()
+              .includes(
+                element.toLowerCase() || topic.toLowerCase().includes(element.toLowerCase())
+              )
+          ) {
+            return true;
+          }
+          return false;
+        });
+        if (contains === true) {
+          return new FilterCardView(room);
```

Comment:
Does the room-list properly update if you toggle safe-search on and off? My guess is no and it might be tricky to force Hydrogen to re-render here since we're using a `ListView`

Reply:
No, the room list doesn't update. How would setting the boolean on the room object enable an update in the list? I don't understand

Reply:
I don't think it would directly but we probably need to do something like that. The views listen to changes that are emitted.

We probably need to make a `class Room extends EventEmitter` [similar to what Hydrogen does](https://github.com/vector-im/hydrogen-web/blob/a565d2a7d377b97a9d02b850311aaaa0b9f88283/src/matrix/room/BaseRoom.js) so we can update the the `isNsfw` property and emit an update for the view to listen and update on.

Reply:
I create a class Room extends Eventmitter in my last [commit](https://github.com/matrix-org/matrix-public-archive/pull/155/commits/11ca940fd22c82fa019e2dd468f3de7148f782ec) but I still can't figure out how the changes will be emitted between the class and the view

Reply:
There's also this [issue from Hydrogen](https://github.com/vector-im/hydrogen-web/issues/727) that I think relates to our problem

Reply:
@Jem256 I'm no Hydrogen expert to be able to explain or know exactly what to do here. I'd just have to hack on it myself in order to give any guidance. You might get some good notes by asking in [Hydrogen room on Matrix](https://matrix.to/#/!nFhKyoXMWWhcbXdKcs:matrix.org?via=matrix.org&via=element.io&via=t2l.io) or checking out the Hydrogen code base to find a similar pattern and how they solved it.